File: 88-merge-sorted-array/88-merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        //used pointer1 and pointer2, interate backwards instead of forwards, and replace element of nums1 in place. 
        int p1 = m - 1, p2 = n -1;
        for (int i = nums1.length - 1; i >= 0; i--) {
            if (p1 >= 0 && p2 >= 0) {
                nums1[i] = (nums1[p1] > nums2[p2]) ? nums1[p1--] : nums2[p2--];
            }
            else if (p1 >= 0) {
                nums1[i] = nums1[p1--];
            }
            else {
                nums1[i] = nums2[p2--];
            }
        }
    }
}

File: 88-merge-sorted-array/88-merge-sorted-array.py
class Solution(object):
    def merge(self, nums1, m, nums2, n):
        """
        :type nums1: List[int]
        :type m: int
        :type nums2: List[int]
        :type n: int
        :rtype: None Do not return anything, modify nums1 in-place instead.
        """
        p1, p2 = m-1, n-1
        for i in reversed(range(len(nums1))):
            if p2 < 0: break
            if p1 >= 0 and nums1[p1] > nums2[p2]:
                nums1[i] = nums1[p1]
                p1 -= 1
            else:
                nums1[i] = nums2[p2]
                p2 -= 1